### ISSUE TITLE
Naming cells, rows and columns

### ISSUE THREAD

**Issue by Enivex** [lvjr/tabularray#381]
I like to use the new interface, where cell properties are handled outside the table itself. Currently (unless I'm missing something), it is only possible to do so using explicit numbers, the letters U-Z for the final rows, and odd/even. This is extremely tedious if you want to change specific cells or rows and make changes that shift the numbering around (you have to update everything).

What I'm looking for is something like
\TblrLabel{clever_cell_name}
that I can put inside the cell and then be able to write

column{clever_cell_name} = {preto=\textbullet\,}
or
row{clever_cell_name} = {font = \bfseries}
or
cell{clever_cell_name} = {r=2,c=1}{}

**Comment by lvjr** [comment]
It is possible to provide `\SetId` and `\SetClass` commands (similar to those in html/css). With `\SetId{Hello}` at the beginning of one cell, its `rownum` and `colnum` are saved and you could write
```latex
row{Hello} = {...}
column{Hello} = {...}
```
or
```latex
row{2-Hello} = {...}
column{Hello-Z} = {...}
```
 With `\SetClass{World}` at the beginning of some cells, their `rownum`s and `colnum`s are saved and you could write
```latex
row{class{World}} = {...}
column{class{World}} = {...}
```
where `class` is a new child selector (see also https://github.com/lvjr/tabularray/issues/382).

The names of id and class must start with capital letters and the names of selectors must start with lowercase letters, so that we can easily distinguish them.

**Comment by Enivex** [comment]
That would be great! Although I have some questions:

Would you need to write cell{Hello}{Hello} to refer to the specific cell, or would it be possible to have the less verbose cell{Hello}?

This distinction becomes more important for the class, because if I add cells (1,2) and (2,1) to World then presumably cell{class{World}}{class{World}} would affect cells (1,2), (2,1), (1,1) and (2,2), while you'd really like cell{class{World}} to affect only (1,2) and (2,1).

Hopefully I explained myself clearly.

**Comment by lvjr** [comment]
Yes, it seems we need to make `tabularray` accept syntax like this
```latex
cell{{1}{2},{2}{1},{3}{4},{4}{3}}
```
or
```latex
cell{{1,2},{2,1},{3,4},{4,3}}
```

**Comment by Enivex** [comment]
cell{{1,2},{2,1},{3,4},{4,3}}

Looks nicest to my eyes, and aligns the closest with usual notation for sets.

**Comment by lvjr** [comment]
The interfaces will be `\SetChild{id=Hello}` or `\SetChild{class=world}`.

The formar will define three expandable child names (see https://github.com/lvjr/tabularray/issues/578):
 
- `Hello` (which expands to a value like `{1}{2}`);
- `Helloh` (which expands to a value like `1`);
- `Hellov`  (which expands to a value like `2`).

Therefore you can write `row{Helloh}`, `column{Hellov}`, or `cell{Hello}`.

And the latter will define three expandable child selectors (see https://github.com/lvjr/tabularray/issues/579):
 
- `world` (which expands to a value like `{1}{2},{2}{1},{3}{4},{4}{3}`);
- `worldh` (which expands to a value like `1,2,3,4`);
- `worldv` (which expands to a value like `2,1,4,3`).

Therefore you can write `row{worldh}`, `column{worldv}`, or `cell{world}`.

The names of ids and child names must start with uppercase letters and the names of classes and child selectors must start with lowercase letters, so that we can easily distinguish them.

**Comment by lvjr** [comment]
As the first step, we need to support syntax `cell{{1}{2},{2}{1},{3}{4},{4}{3}}`.

**Comment by lvjr** [comment]
I think it would be better to use local definitions for ids (indexers) and classes (selectors), so that different `tabularray` tables can share the same id or class names.

**Comment by lvjr** [comment]
Also we can support diagonal selection with `cell{{1}{1}-{5}{5}}`.

**Comment by lvjr** [comment]
Now you can set cell specs in items of tuples:
```tex
\documentclass{article}
\usepackage{tabularray}
\usepackage{xcolor}
\begin{document}
\ExplSyntaxOn
\NewTblrChildIndexer{Hello}
  {
    \tl_set:Nn \lTblrChildIndexTl {{5}{5}}
  }
\NewTblrChildSelector{world}
  {
    \clist_set:Nn \lTblrChildClist {{1}{2},{2}{1},{3}{4},{4}{3}}
  }
\ExplSyntaxOff

\begin{tblr}{
  hline{1,Z},
  cell{{2}{4},{7}{3}} = {bg=blue7},
  cell{{1}{1}-{4}{4},{5}{8}-{8}{5}} = {bg=red7},
  cell{{4}{3}-{1}{1},{6}{5}-{4}{8}} = {bg=green7},
}
  1	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  2	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  3	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  4	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  5	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  6	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  7	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  8	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
\end{tblr}

\begin{tblr}{
  hline{1,Z},
  cell{{7}{6},Hello,{8}{7},world,{6}{8}} = {bg=yellow7},
}
  1	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  2	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  3	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  4	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  5	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  6	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  7	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
  8	& 2	& 3	& 4	& 5	& 6	& 7 & 8 \\
\end{tblr}

\end{document} 
```
![Image](https://github.com/user-attachments/assets/01ae30b5-b162-4fe8-bff1-3c2fc611031b)

**Comment by lvjr** [comment]
I have added `\SetChild` command:
```tex
\documentclass{article}
\usepackage{tabularray}
\usepackage{xcolor}
\begin{document}

\begin{tblr}{
  hline{1,Z},
  cell{Foo,foo,Qux,qux} = {cmd=\fbox},
  row{Barh,barh,Quxh,quxh} = {bg=azure7},
  column{Bazv,bazv,Quxv,quxv} = {fg=red3}
}
  1 & 2 & 3 & \SetChild{id=Foo} 4 & 5 & 6 & 7 & 8 \\
  2 & 2 & \SetChild{idh=Bar} 3 & 4 & 5 & 6 & 7 & 8 \\
  3 & \SetChild{idv=Baz} 2 & 3 & 4 & 5 & 6 & 7 & 8 \\
  \SetChild{id*=Qux} 4 & 2 & 3 & 4 & 5 & 6 & 7 & 8 \\
  5 & 2 & 3 & 4 & \SetChild{class=foo}5 & 6 & 7 & 8 \\
  6 & 2 & 3 & 4 & 5 & \SetChild{classh=bar}6 & 7 & 8 \\
  7 & 2 & 3 & 4 & 5 & 6 & \SetChild{classv=baz}7 & 8 \\
  8 & 2 & 3 & 4 & 5 & 6 & 7 & \SetChild{class*=qux} 8 \\
\end{tblr}

\begin{tblr}{
  hline{1,Z},
  cell{foo,qux} = {cmd=\fbox},
  row{barh,quxh} = {bg=azure7},
  column{bazv,quxv} = {fg=red3}
}
  1 & 2 & 3 & \SetChild{class*=foo} 4 & 5 & 6 & 7 & 8 \\
  2 & 2 & \SetChild{classh=bar} 3 & 4 & 5 & 6 & 7 & 8 \\
  3 & \SetChild{classv=baz} 2 & 3 & 4 & 5 & 6 & 7 & 8 \\
  \SetChild{class=qux} 4 & 2 & 3 & 4 & 5 & 6 & 7 & 8 \\
  5 & 2 & 3 & 4 & \SetChild{class=foo}5 & 6 & 7 & 8 \\
  6 & 2 & 3 & 4 & 5 & \SetChild{classh=bar}6 & 7 & 8 \\
  7 & 2 & 3 & 4 & 5 & 6 & \SetChild{classv=baz}7 & 8 \\
  8 & 2 & 3 & 4 & 5 & 6 & 7 & \SetChild{class*=qux} 8 \\
\end{tblr}

\end{document} 
```

![Image](https://github.com/user-attachments/assets/f28f1f13-0003-4d7b-92a7-b7c0940ebc22)

**Comment by Enivex** [comment]
This looks great!
